packages/esm-patient-medications-app/src/add-drug-order/drug-search/drug-search.component.tsx

```diff
@@ -8,15 +8,20 @@ import styles from './order-basket-search.scss';
 
 export interface DrugSearchProps {
   openOrderForm: (searchResult: DrugOrderBasketItem) => void;
+  promptBeforeClosing: (testFcn: () => boolean) => void;
 }
 
-export default function DrugSearch({ openOrderForm }: DrugSearchProps) {
+export default function DrugSearch({ openOrderForm, promptBeforeClosing }: DrugSearchProps) {
   const { t } = useTranslation();
   const isTablet = useLayoutType() === 'tablet';
   const [searchTerm, setSearchTerm] = useState('');
   const debouncedSearchTerm = useDebounce(searchTerm);
   const searchInputRef = useRef(null);
 
+  useEffect(() => {
+    promptBeforeClosing(() => !!searchTerm);
+  }, [searchTerm]);
```

Comment:
```suggestion
  useEffect(() => {
    promptBeforeClosing(() => false);
  }, []);
```

It's a search box, there's not really anything to save there.